vtex-sites/nextjs.store
# chore: Suspend useQueries to improve page performance

## What's the purpose of this pull request?
Improve lighthouse metrics, the main focus is to decrease the home page Total Blocking Time (TBT)

## How does it work?
Profiling the home page and analyzing the flame graphs, I noticed components with a huge percentage of the TBT during the hydration phase. Take a look at these screenshots from the same profile.

![image](https://user-images.githubusercontent.com/15680320/167179321-e364e2e7-d362-4978-8e07-2f2aeb040277.png)

**First problem**: As you can see the usePersonQuery used by the ButtonSignIn, is taking considerable time, and it's basically to compile code from useSWR code. So, to tackle this problem, could remove this compilation from this task and put into a new task, and by doing this we decrease the TBT.

Also, the useSWR hook is used by the `ProductShelf` and `ProductTiles` via`useProductsQuery`, and this hook adds some computation to make the fetch itself in the end of this huge task that I mentioned before. You can see at this 4 screenshots, for each swr execution (usePersonQuery, and 3 times for useProductsQuery)

![image](https://user-images.githubusercontent.com/15680320/167183687-5ad85d45-94ef-41c6-863c-1487ec672085.png)
![image](https://user-images.githubusercontent.com/15680320/167184343-0e563d1e-9786-44ab-ac91-82a3ffd80c33.png)
![image](https://user-images.githubusercontent.com/15680320/167184436-fed2f081-116e-44cf-a457-d35eb5b788d6.png)
![image](https://user-images.githubusercontent.com/15680320/167184806-7a94c65d-dcb9-482b-9bd4-469c0e499b13.png)

So, following the same strategy from the first problem, this PR suspends the `useProductsQuery` strategy.

After these changes, these components are being rendered in a new task, different from the hydration (this is not the hydration itself, but the first render), but, also, the end of the hydration doesn't have all that fetches in the flame graph.

![image](https://user-images.githubusercontent.com/15680320/167187904-3150941a-21a2-4534-a5f1-80323da9b69a.png)

## How to test it?
Navigate on pages, and check if the behavior is maintained: shelves skeletons, login button...

## References
https://reactjs.org/docs/react-api.html#reactsuspense-in-server-side-rendering
https://swr.vercel.app/docs/suspense

## Checklist

<em>You may erase this after checking them all ;)</em>



- PR description
